lib/Remotery.c

```diff
@@ -3665,9 +3675,12 @@ static rmtError ThreadSampler_Constructor(ThreadSampler* thread_sampler)
     thread_sampler->next = NULL;
 
     // Set the initial name to Thread0 etc.
-    thread_sampler->name[0] = 0;
-    strncat_s(thread_sampler->name, sizeof(thread_sampler->name), "Thread", 6);
-    itoahex_s(thread_sampler->name + 6, sizeof(thread_sampler->name) - 6, AtomicAdd(&countThreads, 1));
+    GetDebuggerThreadName(thread_sampler->name);
```

Comment:
Yes. I'm maintaining an app with over 33 threads (ugh). And to mange it well, I have named each thread. I do this at thread creation so they are set before this code in Remotery gets called.

When doing profiling work, it's vital to know what thread is hungry at a glance, since there is no way to easily correlate the name ThreadXwith the actual thread using it.